builder/main.py

```diff
@@ -148,11 +167,14 @@ def _ldflags_for_hex(env, ldflags):
 
 elif "stlink" in upload_protocol:
     mcu = board_config.get("build.mcu")
+    # either derive value for the "part" switch from MCU name, or use the value 
+    # in the board manifest, in cases in which the derivation would be wrong.
+    flash_target = board_config.get("upload.stm8flash_target", mcu[:8] + "?" + mcu[9])
     env.Replace(
         UPLOADER="stm8flash",
         UPLOADERFLAGS=[
             "-c", "$UPLOAD_PROTOCOL",
-            "-p", "%s" % mcu[:8] + "?" + mcu[9],
+            "-p", "%s" % flash_target,
```

Comment:
Do we really need to format it here? It looks like the `flash_target` is already a string.

Reply:
Unnecessary string format removed in latest commit.